lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +237,61 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Package(Requirement):
+    def __init__(self, name_string, version_string=None):
+        if version_string is not None:
+            if version_string[0].isdigit():
+                name_string = "%s ==%s" % (name_string, version_string)
+            else:
+                name_string = "%s %s" % (name_string, version_string)
+        super(Package, self).__init__(name_string)
+
+    def have_version_specifier(self):
+        if len(self.specs) == 0:
+            return False
+        return True
+
+    def get_name(self):
+        return self.name.lower()
+
+    def satisfied_by(self, version_to_test):
+        return self.specifier.contains(version_to_test)
+
+    def install_command(self):
+        return str(self)
+
+
+def _is_valid_pkg_name(name):
+    try:
+        Requirement(name)
+        return True
+    except RequirementParseError:
+        return False
+
+
+def _fetch_one_pkg(name):
+    stack = list()
+    for ele in name:
+        if _is_valid_pkg_name(ele):
+            if len(stack) > 0:
+                yield ",".join(stack)
+            stack = [ele]
+        else:
+            stack.append(ele)
+    yield ",".join(stack)
+
+
+def _parses_name_input(module, name):
+    """Parse name list to package list."""
+    package_list = list()
+    try:
+        for package_name in _fetch_one_pkg(name):
```

Comment:
You don't need to create a `list()` and then mutate it. Instead, you can turn it into a list comprehension, making the whole function shorter and better readable:
```python
try:
    return [Package(name) for name in _recover_package_names(names)]
except Exception as e:  #  includes RequirementParseError, but also other errors, which might occur, and would need fail_json anyway
    module.fail_json(msg=str(e))
```